### PR TITLE
=doc Consolidate Routing DSL testkit docs #1290

### PR DESCRIPTION
**Pull request by atiqsayyed**
Ref #1290
Merged routing dsl testkit

### CODE REVIEW

**Comment by jlprat** [comment]
OK TO TEST

File: docs/src/main/paradox/scala/http/routing-dsl/testkit.md [hunk 1]
```diff
@@ -96,19 +170,28 @@ implicit rejection and exception handlers in scope.
 Since the testkit is request-based, you cannot test requests that are illegal or impossible in HTTP. One
 such instance is testing a route that begins with the `pathEnd` directive, such as `routeFragment` here:
 
-@@snip [TestKitFragmentSpec.scala]($test$/scala/docs/http/scaladsl/server/TestKitFragmentSpec.scala) { #fragment }
+Scala
+:   @@snip [TestKitFragmentSpec.scala]($test$/scala/docs/http/scaladsl/server/TestKitFragmentSpec.scala) { #fragment }
```
Comment (raboof) [inline docs/src/main/paradox/scala/http/routing-dsl/testkit.md]: no Java fragment for this one?

File: docs/src/main/paradox/scala/http/routing-dsl/testkit.md [hunk 2]
```diff
@@ -77,17 +80,88 @@ The following inspectors are defined:
...
 with HTTP response parts or rejections. Sometimes, however, you will want to verify that your service also translates
 @ref[Rejections](rejections.md) to HTTP responses in the way you expect.
 
-You do this by wrapping your route with the `akka.http.scaladsl.server.Route.seal`. The `seal` wrapper applies the logic of the in-scope @ref[ExceptionHandler](exception-handling.md) and
+You do this by wrapping your route with the @scala[`akka.http.scaladsl.server.Route.seal`]@java[`akka.http.javadsl.server.Route.seal`]. The `seal` wrapper applies the logic of the in-scope @ref[ExceptionHandler](exception-handling.md) and
 @ref[RejectionHandler](rejections.md#the-rejectionhandler) to all exceptions and rejections coming back from the route,
 and translates them to the respective `HttpResponse`.
 
-Note that explicit call on the `akka.http.scaladsl.server.Route.seal` method is needed in test code, but in your application code it is not necessary.
+Note that explicit call on the @scala[`akka.http.scaladsl.server.Route.seal`]@java[`akka.http.javadsl.server.Route.seal`] method is needed in test code, but in your application code it is not necessary.
```
Comment (jonas) [inline docs/src/main/paradox/scala/http/routing-dsl/testkit.md]: I suggest we simplify this to `Note that explicit call on the `Route.seal()` method ...`. Most other places do not use the fully qualified name when referring to methods.

File: docs/src/main/paradox/scala/http/routing-dsl/testkit.md [hunk 3]
```diff
@@ -77,17 +80,88 @@ The following inspectors are defined:
...
+@@@
+
 ## Testing sealed Routes
 
 The section above describes how to test a "regular" branch of your route structure, which reacts to incoming requests
 with HTTP response parts or rejections. Sometimes, however, you will want to verify that your service also translates
 @ref[Rejections](rejections.md) to HTTP responses in the way you expect.
 
-You do this by wrapping your route with the `akka.http.scaladsl.server.Route.seal`. The `seal` wrapper applies the logic of the in-scope @ref[ExceptionHandler](exception-handling.md) and
+You do this by wrapping your route with the @scala[`akka.http.scaladsl.server.Route.seal`]@java[`akka.http.javadsl.server.Route.seal`]. The `seal` wrapper applies the logic of the in-scope @ref[ExceptionHandler](exception-handling.md) and
```
Comment (jonas) [inline docs/src/main/paradox/scala/http/routing-dsl/testkit.md]: Java does not have the notion of a "in-scope ExceptionHandler", so we need to carry over the Java version here:

I suggest something like this:
```
You do this by calling the `Route.seal()` method. The `Route.seal()` method applies the logic of the @scala[in-scope]
@ref[RejectionHandler](rejections.md#the-rejectionhandler) and @ref[RejectionHandler](rejections.md#the-rejectionhandler)
@java[passed as method arguments] to all exceptions and rejections coming back from the 
route, and translates them to the respective `HttpResponse`.
```

File: docs/src/main/paradox/scala/http/routing-dsl/testkit.md [hunk 4]
```diff
@@ -77,17 +80,88 @@ The following inspectors are defined:
 which always produces absolute URIs for incoming request based on the request URI and the `Host`-header of
 the request. You can customize this behavior by bringing a custom instance of `DefaultHostInfo` into scope.
 
+@@@
+
+@@@ div { .group-java }
+
+akka-http has a testkit that provides a convenient way of testing your routes with JUnit. It allows
+running requests against a route (without hitting the network) and provides means to assert against
+response properties in a compact way.
```
Comment (jonas) [inline docs/src/main/paradox/scala/http/routing-dsl/testkit.md]: I suggest this section is removed and we use the same introduction as the Scala version. The following text would make sense inside the `## Usage` section defined after the intro.

File: docs/src/main/paradox/scala/http/routing-dsl/testkit.md [hunk 5]
```diff
@@ -3,16 +3,24 @@
...
-"com.typesafe.akka" %% "akka-http-testkit" % "$project.version$"
-```
+sbt
+:   @@@vars
+    ```
+    "com.typesafe.akka" %% "akka-http-testkit" % "$project.version$"
+    ```
+    @@@
+
 @@@
```
Comment (jonas) [inline docs/src/main/paradox/scala/http/routing-dsl/testkit.md]: This should be included in both the Java and Scala version and ideally should also show Gradle and Maven snippets:

```
sbt
:   @@@vars
    ```scala
    "com.typesafe.akka" %% "akka-http-testkit" % "$project.version$" $crossString$
    ```
    @@@

Gradle
:   @@@vars
    ```
    compile group: 'com.typesafe.akka', name: 'akka-http-testkit_$scala.binary_version$', version: '$project.version$'
    ```
    @@@

Maven
:   @@@vars
    ```xml
    <dependency>
      <groupId>com.typesafe.akka</groupId>
      <artifactId>akka-http-testkit_$scala.binary_version$</artifactId>
      <version>$project.version$</version>
    </dependency>
    ```
    @@@
```

**Comment by ktoso** [comment]
LGTM, thanks!

**Pull request merged**